src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -110,15 +112,16 @@ protected function extractAttributes($object, $format = null, array $context = a
      */
     protected function getAttributeValue($object, $attribute, $format = null, array $context = array())
     {
-        if (null !== $this->classDiscriminatorResolver) {
-            $mapping = $this->classDiscriminatorResolver->getMappingForMappedObject($object);
-
-            if (null !== $mapping && $attribute == $mapping->getTypeProperty()) {
-                return $this->classDiscriminatorResolver->getTypeForMappedObject($object);
+        $cacheKey = spl_object_hash($object);
+        if (!array_key_exists($cacheKey, $this->discriminatorCache)) {
```

Comment:
using `if (false === $this->discriminatorCache[$cacheKey] ?? false)` will even be faster

Reply:
Indeed, can’t we store false instead of null? If it makes a significant difference ofc

Reply:
I tried to store `false` and to test against `true`, but this breaks the cache.
```php
if (true === ($this->discriminatorCache[$cacheKey] ?? true)) {
            $this->discriminatorCache[$cacheKey] = false;
            // ...
        }
```